pkg/resources/account.go

```diff
@@ -20,20 +19,19 @@ import (
 
 var accountSchema = map[string]*schema.Schema{
 	"name": {
-		Type:         schema.TypeString,
-		Required:     true,
-		Description:  "Specifies the identifier (i.e. name) for the account; must be unique within an organization, regardless of which Snowflake Region the account is in. In addition, the identifier must start with an alphabetic character and cannot contain spaces or special characters except for underscores (_). Note that if the account name includes underscores, features that do not accept account names with underscores (e.g. Okta SSO or SCIM) can reference a version of the account name that substitutes hyphens (-) for the underscores.",
-		ValidateFunc: snowflakeValidation.ValidateAccountIdentifier,
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "Specifies the identifier (i.e. name) for the account; must be unique within an organization, regardless of which Snowflake Region the account is in. In addition, the identifier must start with an alphabetic character and cannot contain spaces or special characters except for underscores (_). Note that if the account name includes underscores, features that do not accept account names with underscores (e.g. Okta SSO or SCIM) can reference a version of the account name that substitutes hyphens (-) for the underscores.",
 		// Name is automatically uppercase by Snowflake
 		StateFunc: func(val interface{}) string {
 			return strings.ToUpper(val.(string))
 		},
+		ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
```

Comment:
Now I'm confused: shouldn't it be `IsValidAccountIdentifier`?

Reply:
From the PR description:
`replace ValidateAccountIdentifier with already existing IsValidIdentifier[sdk.AccountObjectIdentifier]()`

Reason: this name is handled as AccountObjectIdentifier in the resource methods.

Reply:
It can be confusing because it's account object, so from <organization_name>.<account_name> it's the second part. As we don't have a dedicated validator for only account names this one was used.